recipes/coin-cgl/all/conanfile.py

```diff
@@ -79,70 +72,72 @@ def validate(self):
-                    "LD": "{} link -nologo".format(tools.unix_path(self._user_info_build["automake"].compile)),
-                    "AR": "{} lib".format(tools.unix_path(self._user_info_build["automake"].ar_lib)),
-                }
-                with tools.environment_append(env):
-                    yield
-        else:
-            yield
-
-    def _configure_autotools(self):
-        if self._autotools:
-            return self._autotools
-        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
-        self._autotools.libs = []
-        yes_no = lambda v: "yes" if v else "no"
-        configure_args = [
-            "--enable-shared={}".format(yes_no(self.options.shared)),
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+
+        tc = AutotoolsToolchain(self)
+        tc.configure_args.extend([
             "--without-blas",
             "--without-lapack",
-        ]
-        if self.settings.compiler == "Visual Studio":
-            self._autotools.cxx_flags.append("-EHsc")
-            configure_args.append(f"--enable-msvc={self.settings.compiler.runtime}")
-            if Version(self.settings.compiler.version) >= 12:
-                self._autotools.flags.append("-FS")
-        self._autotools.configure(configure_dir=os.path.join(self.source_folder, self._source_subfolder), args=configure_args)
-        return self._autotools
+        ])
+        if is_msvc(self):
+            tc.extra_cxxflags.append("-EHsc")
+            tc.configure_args.append(f"--enable-msvc={msvc_runtime_flag(self)}")
+            if check_min_vs(self, "180", raise_invalid=False):
+                tc.extra_cflags.append("-FS")
+                tc.extra_cxxflags.append("-FS")
+        env = tc.environment()
+        if is_msvc(self):
+            compile_wrapper = unix_path(self, self.conf.get("user.automake:compile-wrapper", check_type=str))
+            ar_wrapper = unix_path(self, self.conf.get("user.automake:lib-wrapper", check_type=str))
+            env.define("CC", f"{compile_wrapper} cl -nologo")
+            env.define("CXX", f"{compile_wrapper} cl -nologo")
+            env.define("LD", f"{compile_wrapper} link -nologo")
+            env.define("AR", f"{ar_wrapper} \"lib -nologo\"")
+            env.define("NM", "dumpbin -symbols")
+            env.define("OBJDUMP", ":")
+            env.define("RANLIB", ":")
+            env.define("STRIP", ":")
+        if self._settings_build.os == "Windows":
+            # TODO: Something to fix in conan client or pkgconf recipe?
+            # This is a weird workaround when build machine is Windows. Here we have to inject regular
+            # Windows path to pc files folder instead of unix path flavor injected by AutotoolsToolchain...
+            env.define("PKG_CONFIG_PATH", self.generators_folder)
```

Comment:
@jcar87 There's some suspicious code 🏫

Reply:
related issue (for coin-osi): https://github.com/conan-io/conan/issues/13098